nucleus/payara-modules/healthcheck-core/src/main/java/fish/payara/nucleus/healthcheck/HistoricHealthCheckEventStore.java

```diff
@@ -68,20 +67,17 @@
     @Inject
     private ServerEnvironment serverEnv;
 
-    private HazelcastInstance instance;
-
-    private SortedSet<HistoricHealthCheckEvent> historicStore;
+    private NavigableSet<HistoricHealthCheckEvent> historicStore;
 
     void initialize(int storeSize) {
-        historicStore = Collections.synchronizedSortedSet(new BoundedTreeSet<HistoricHealthCheckEvent>(storeSize));
+        historicStore = new BoundedTreeSet<>(storeSize);
 
         if (hzCore.isEnabled()) {
-            instance = hzCore.getInstance();
+            HazelcastInstance instance = hzCore.getInstance();
```

Comment:
Can you use the ClusteredStore service here rather than the HazelcastInstance directly?
```
@Inject
ClusteredStore store
```